File: C#/LongestHarmoniousSubsequence.cs
  
//Ian Hays
//02-04-2021
//https://leetcode.com/problems/longest-harmonious-subsequence/
//SC: O(N) TC: O(N)

public class Solution {
    public int FindLHS(int[] nums) {
        IDictionary<int,int> dict = new Dictionary<int,int>();
        foreach(int num in nums) 
            dict[num] = dict.ContainsKey(num) ? dict[num] + 1 : 1;
        return dict.Max(d => dict.ContainsKey(d.Key + 1) ? d.Value + dict[d.Key + 1]: 0);
    }
}

File: C++/LongestHarmoniousSubsequence.cpp
//Ian Hays
//02-04-2021
//https://leetcode.com/problems/longest-harmonious-subsequence/
//SC: O(N) TC: O(N)

class Solution {
public:
    int findLHS(vector<int>& nums) {
        unordered_map<int,int> umap;
        for(auto num: nums) 
            umap[num]++;
        
        int res = 0;
        for(auto [num, freq] : umap){
            if(umap.find(num+1) != umap.end())
                res = max(res, freq + umap.at(num+1));
        }
        return res;
    }
};
